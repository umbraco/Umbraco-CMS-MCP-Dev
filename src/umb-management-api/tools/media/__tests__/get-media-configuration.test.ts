import GetMediaConfigurationTool from "../get/get-media-configuration.js";
import { jest } from "@jest/globals";

describe("get-media-configuration", () => {
  let originalConsoleError: typeof console.error;

  beforeEach(() => {
    originalConsoleError = console.error;
    console.error = jest.fn();
  });

  afterEach(() => {
    console.error = originalConsoleError;
  });

  it("should get media configuration", async () => {
    const result = await GetMediaConfigurationTool().handler({}, { signal: new AbortController().signal });
    expect(result).toBeDefined();
    expect(result.content).toBeDefined();
    expect(result.content.length).toBeGreaterThan(0);
    expect(result).toMatchSnapshot();
  });
}); 