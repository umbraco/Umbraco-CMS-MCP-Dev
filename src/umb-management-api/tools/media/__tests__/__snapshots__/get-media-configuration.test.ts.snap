// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-media-configuration should get media configuration 1`] = `
{
  "content": [
    {
      "text": "{"disableDeleteWhenReferenced":false,"disableUnpublishWhenReferenced":false,"reservedFieldNames":["ContentType","Key","Properties","Id","Name","UrlSegment","SortOrder","Level","Path","TemplateId","CreatorId","CreateDate","WriterId","UpdateDate","Cultures","ItemType","Parent","Children","get_ContentType","get_Key","get_Properties","GetProperty","get_Id","get_Name","get_UrlSegment","get_SortOrder","get_Level","get_Path","get_TemplateId","get_CreatorId","get_CreateDate","get_WriterId","get_UpdateDate","get_Cultures","get_ItemType","get_Parent","IsDraft","IsPublished","get_Children"]}",
      "type": "text",
    },
  ],
}
`;
